### ISSUE TITLE
-auto-orient option may flip images in older Imagemagick versions

### ISSUE THREAD

**Issue by risvh** [jasonrohrer/minorGems#9]
### TLDR:

The auto-orient option is only required for Imagemagick 6.9.10.75 or newer. For older versions, having the -auto-orient option flips the resulting TGA images.
 
 
### The Rest of the Story:

This change https://github.com/jasonrohrer/minorGems/commit/1a4ab82d3afe13b83e2f641b34b4d7b99f68f23d was made to fix this issue https://github.com/jasonrohrer/OneLife/issues/848; the issue has been reported numerous times in the github page of Imagemagick:

https://github.com/ImageMagick/ImageMagick/issues/3061
https://github.com/ImageMagick/ImageMagick/issues/3844
https://github.com/ImageMagick/ImageMagick/issues/5148
https://github.com/ImageMagick/ImageMagick/issues/5973

The same issue has been fixed (de9049d978784c1b91cf9d03e335c0b5991ea42c) over in 2HOL but then retracted (c95828737d29cdb4d93c69fd6a63af83daba4aa5) because of this same reason - older Imagemagick versions are not uncommon. Many in 2HOL use WSL to compile the game. I'd say half of them have the version 6.9.10-23 that comes with the distro, the other half have 6.9.11-60.

I have done some tests with the versions of Imagemagick available here. (https://imagemagick.org/archive/releases/) The image flip happens in version 6.9.10.75 or newer.

Below is 2HOL's fix to add the auto-orient option conditional on the Imagemagick version the environment has. (This sed takes place in a repo which contains some alternative compile scripts for the game, see the bottom)

```#New versions of imagemagick flips images after conversion
im_version=`convert -list configure | sed '/^LIB_VERSION_NUMBER */!d; s//,/;  s/,/,0/g;  s/,0*\([0-9][0-9]\)/\1/g' | head -n 1`
if [ "$im_version" -ge "06091075" ]; then
	sed -i 's/convert $< -type truecolormatte $@/convert -auto-orient $< -type truecolormatte $@/' ../minorGems/game/platforms/SDL/Makefile.all
fi
```

Related changes:
https://github.com/risvh/miniOneLifeCompile/commit/39fc2f0ef5b97815a42f6b1d4c79cd7d31a5d1ec
https://github.com/risvh/miniOneLifeCompile/commit/d2443ed5aa4383928ae241ace7d1eaa507567c04
https://github.com/risvh/miniOneLifeCompile/commit/26f1a511fd87c4cbee544f5d946a3550e0bd2057

**Comment by jasonrohrer** [comment]
Hmm... yeah, this is a really unfortunate issue.

I'm guessing that this can be done inside the makefile itself, and not with a script that modifies the makefile.  Maybe something like this:

https://stackoverflow.com/questions/3728372/version-number-comparison-inside-makefile
